Sources/_OpenAPIGeneratorCore/Parser/validateDoc.swift

```diff
@@ -14,6 +14,71 @@
 
 import OpenAPIKit
 
+/// Validates all content types from an OpenAPI document represented by a ParsedOpenAPIRepresentation.
+///
+/// This function iterates through the paths, endpoints, and components of the OpenAPI document,
+/// checking and reporting any invalid content types using the provided validation closure.
+///
+/// - Parameters:
+///   - doc: The OpenAPI document representation.
+///   - validate: A closure to validate each content type.
+/// - Throws: An error with diagnostic information if any invalid content types are found.
+func validateContentTypes(in doc: ParsedOpenAPIRepresentation, validate: (String) -> Bool) throws {
+    for (path, pathValue) in doc.paths {
+        guard case .b(let pathItem) = pathValue else { continue }
+        for endpoint in pathItem.endpoints {
+
+            if let eitherRequest = endpoint.operation.requestBody {
+                if case .b(let actualRequest) = eitherRequest {
+                    for contentType in actualRequest.content.keys {
+                        if !validate(contentType.rawValue) {
+                            throw Diagnostic.error(
+                                message:
+                                    "Invalid content type string: '\(contentType.rawValue)' found in requestBody at path '\(path.rawValue)'. Must have 2 components separated by a slash '<type>/<subtype>'.\n"
+                            )
+                        }
+                    }
+                }
+            }
+
+            for eitherResponse in endpoint.operation.responses.values {
+                if case .b(let actualResponse) = eitherResponse {
+                    for contentType in actualResponse.content.keys {
+                        if !validate(contentType.rawValue) {
+                            throw Diagnostic.error(
+                                message:
+                                    "Invalid content type string: '\(contentType.rawValue)' found in responses at path '\(path.rawValue)'. Must have 2 components separated by a slash '<type>/<subtype>'.\n"
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    for component in doc.components.requestBodies.values {
```

Comment:
Let's iterate both keys and values, to allow us to print the key later down.
```suggestion
    for (key, component) in doc.components.requestBodies {
```